### PR TITLE
chore: make TS enums strictly PascalCase

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
TypeScript enum names and keys should always be PascalCase, which is the commonly accepted best practice (see the official docs: https://www.typescriptlang.org/docs/handbook/enums.html). This PR does the following:
- Adds ESLint rules to enforce both PascalCase for the enum names and their respective keys.
- Fix the whole FS codebase, including `superset-ui` packages/plugins. The enum values have been left in tact so as to not cause a breaking change.

Note, that I've made a few exceptions, where deviating from pascal case makes more sense, especially the locales and currencies, where it felt dangerous and less intuitive to move to pascal case.

### TESTING INSTRUCTIONS
1. Pull the codebase, try to rename a TS enum to snake case, and notice how the linter complains.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/ChartPills.tsx [hunk 1]
```diff
@@ -22,11 +22,11 @@ import RowCountLabel from 'src/explore/components/RowCountLabel';
...
-enum CHART_STATUS_MAP {
-  failed = 'danger',
-  loading = 'warning',
-  success = 'success',
-}
+const CHART_STATUS_MAP = {
+  failed: 'danger',
+  loading: 'warning',
+  success: 'success',
+};
```
Comment (villebro) [inline superset-frontend/src/explore/components/ChartPills.tsx:29]: This was really not a proper enum or used as such, so I changed it to a regular object.

File: superset-frontend/src/components/Table/cell-renderers/NumericCell/index.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (villebro) [inline superset-frontend/src/components/Table/cell-renderers/NumericCell/index.tsx:1]: I added this single override, as it felt sensible to leave the locales and currencies as-is

**Review by michael-s-molina — CHANGES_REQUESTED** [review]
Thank you for the improvement @villebro. I left some first-pass comments.

File: superset-frontend/src/components/Table/cell-renderers/NumericCell/index.tsx [hunk 3]
```diff
@@ -403,8 +355,57 @@ export enum CurrencyCode {
...
+
+export enum CurrencyDisplay {
+  Symbol = 'symbol',
+  Code = 'code',
+  Name = 'name',
+}
+
 export function NumericCell(props: NumericCellProps) {
-  const { value, locale = LocaleCode.en_US, options } = props;
+  const { value, locale = 'en-US', options } = props;
```
[the remark this answers is not in the document]
Comment (villebro) [reply superset-frontend/src/components/Table/cell-renderers/NumericCell/index.tsx:408]: Thanks for spotting this - this is a remnant from when I was planning on removing the `LocaleCode` enum. I'll fix it back 👍

File: superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts [hunk 4]
```diff
@@ -22,40 +22,40 @@ import logger from './logging';
...
+  EscapeMarkdownHtml = 'ESCAPE_MARKDOWN_HTML',
+  EstimateQueryCost = 'ESTIMATE_QUERY_COST',
+  GenericChartAxes = 'GENERIC_CHART_AXES',
+  GlobalAsyncQueries = 'GLOBAL_ASYNC_QUERIES',
+  HorizontalFilterBar = 'HORIZONTAL_FILTER_BAR',
+  ListviewsDefaultCardView = 'LISTVIEWS_DEFAULT_CARD_VIEW',
+  ScheduledQueries = 'SCHEDULED_QUERIES',
+  ShareQueriesViaKvStore = 'SHARE_QUERIES_VIA_KV_STORE',
+  SqllabBackendPersistence = 'SQLLAB_BACKEND_PERSISTENCE',
+  SqlValidatorsByEngine = 'SQLLAB_BACKEND_PERSISTENCE',
```
[the remark this answers is not in the document]
Comment (villebro) [reply superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts:54]: I had to go through all key+value combos one by one, so this one came up during that exercise..

[the remark this answers is not in the document]
Comment (eschutho) [reply superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts]: Lots of ☕ apparently! :)

[the remark this answers is not in the document]
Comment (michael-s-molina) [reply superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts]: ☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕☕

**Review by michael-s-molina — APPROVED** [review]
LGTM!

**Comment by villebro** [comment]
I'm gonna go ahead and merge this, as it conflicts extremely easily due to the number of touched files. Thanks for the reviews!

**Comment by villebro** [comment]
...but I'll leave it open until the end of the working day PST so anyone else can post commentts if they have time to review!

**Review by kgabryje — APPROVED** [review]
Love it, 100% recommend for bed time read if you have insomnia. LGTM!

**Comment by rusackas** [comment]
You know when you forget to blink for too long, then it hurts when you do?

I'm halfway done reviewing. My browser hates this PR, but I love that you've done it :D

File: superset-frontend/src/SqlLab/reducers/sqlLab.js [hunk 5]
```diff
@@ -332,15 +332,15 @@ export default function sqlLabReducer(state = {}, action) {
     },
     [actions.REQUEST_QUERY_RESULTS]() {
       return alterInObject(state, 'queries', action.query, {
-        state: QueryState.FETCHING,
+        state: QueryState.Fetching,
```
Comment (rusackas) [inline superset-frontend/src/SqlLab/reducers/sqlLab.js:335]: 687474703a2f2f696d616765732e68756666696e67746f6e706f73742e636f6d2f323031342d30362d32352d666574636868617070656e2e676966

**Comment by villebro** [comment]
> I'm halfway done reviewing. My browser hates this PR, but I love that you've done it :D

The fact that this PR is 2k LOC and almost crashes the browser shows just how bad this inconsistency problem had gotten over time. I look forward to doing the same for the backend codebase.. **shudders**

File: superset-frontend/src/explore/components/controls/ConditionalFormattingControl/FormattingPopoverContent.tsx [hunk 6]
```diff
@@ -18,16 +18,16 @@
  */
 import React from 'react';
 import { styled, SupersetTheme, t, useTheme } from '@superset-ui/core';
+import {
+  Comparator,
+  MultipleValueComparators,
+} from '@superset-ui/chart-controls';
```
Comment (rusackas) [inline superset-frontend/src/explore/components/controls/ConditionalFormattingControl/FormattingPopoverContent.tsx:24]: Double checking... is this being pulled from the right place?

Comment (rusackas) [reply superset-frontend/src/explore/components/controls/ConditionalFormattingControl/FormattingPopoverContent.tsx:24]: looks like the import changes... wasn't sure that's intended.

Comment (villebro) [reply superset-frontend/src/explore/components/controls/ConditionalFormattingControl/FormattingPopoverContent.tsx:24]: Yes - both `Comparator` and `MultipleValueComparators` were duplicated, and existed both here and in `superset-ui/core`. In this PR I removed the one here and replaced it with references to `superset-ui/core`.

File: superset-frontend/src/features/reports/types.ts [hunk 7]
```diff
@@ -26,14 +26,14 @@ export type ReportCreationMethod = 'charts' | 'dashboards' | 'alerts_reports';
...
+  Charts = 'charts',
 }
 
-export enum NOTIFICATION_FORMATS {
-  TEXT = 'TEXT',
-  PNG = 'PNG',
-  CSV = 'CSV',
+export enum NotificationFormats {
+  Text = 'TEXT',
+  Png = 'PNG',
```
Comment (rusackas) [inline superset-frontend/src/features/reports/types.ts:35]: Nit, but I'd be A-OK with keeping PNG, CSV, and other acronyms as all-caps ¯\\\_(ツ)_/¯

**Review by rusackas — APPROVED** [review]
LGTM!
_...passes out_

File: superset-frontend/src/features/reports/types.ts (hunk 7 above)
Comment (villebro) [reply superset-frontend/src/features/reports/types.ts]: That's an excellent point, and it's strictly speaking PascalCase - updated!

**Pull request merged**